include/eve/module/core/regular/impl/simd/arm/neon/shuffle_l0.hpp

```diff
@@ -0,0 +1,41 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+
+// NOTE: using no_matching_shuffle_t {} because otherwise apple clang crashes.
+
+template<arithmetic_scalar_value T, typename N, std::ptrdiff_t G, std::ptrdiff_t... I>
+EVE_FORCEINLINE auto
+shuffle_l0_impl_(EVE_SUPPORTS(neon128_), wide<T, N> x, fixed<G>, pattern_t<I...>)
+requires std::same_as<abi_t<T, N>, arm_64_>
+{
+  constexpr std::array idxs {I...};
+  if constexpr( idx::is_identity(idxs) ) return x;
+  else if constexpr( idx::is_zero(idxs) ) return wide<T, N> {0};
+  // So far no l0 shuffles on arm that support 0s, needs l1.
+  else if constexpr ( idx::has_zeroes(idxs) ) return no_matching_shuffle;
+  else if constexpr ( sizeof(T) == 4 )
+  {
+    if constexpr ( idx::matches(idxs, {1, 0}) )
+    {
+      if constexpr( std::same_as<T, float> ) return vrev64_f32(x);
```

Comment:
Again, I assume this is due to simplification ?

Reply:
Well it's size 4, pattern 1, 0 and float. So it's 100% match.